src/NServiceBus.TransactionalSession/TransactionalSessionBase.cs

```diff
@@ -0,0 +1,132 @@
+namespace NServiceBus.TransactionalSession
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Persistence;
+    using Transport;
+
+    abstract class TransactionalSessionBase : ITransactionalSession
+    {
+        protected TransactionalSessionBase(
+            ICompletableSynchronizedStorageSession synchronizedStorageSession,
+            IMessageSession messageSession,
+            IMessageDispatcher dispatcher)
+        {
+            this.synchronizedStorageSession = synchronizedStorageSession;
+            this.messageSession = messageSession;
+            this.dispatcher = dispatcher;
+            pendingOperations = new PendingTransportOperations();
+        }
+
+        public ISynchronizedStorageSession SynchronizedStorageSession
+        {
+            get
+            {
+                if (!IsOpen)
+                {
+                    throw new InvalidOperationException(
+                        "Before accessing the SynchronizedStorageSession, make sure to open the session by calling the `Open`-method.");
+                }
+
+                return synchronizedStorageSession;
+            }
+        }
+
+        public string SessionId => options?.SessionId;
```

Comment:
Should we throw if not open as we do for the storage session property get?

Reply:
👍